okhttp-testing-support/src/main/kotlin/okhttp3/OkHttpClientTestRule.kt

```diff
@@ -76,8 +112,13 @@ class OkHttpClientTestRule : TestRule {
   }
 
   @Synchronized private fun addEvent(event: String) {
-    logger?.info(event)
-    clientEventsList.add(event)
+    if (recordEvents) {
+      logger?.info(event)
+
+      synchronized(clientEventsList) {
```

Comment:
omit this and make everything travel through the log handler?

Reply:
This is tough, the OkHttpClientTestRule works on recording things but not logging immediately, deferring until there is a failure that is actually interesting.  Not sure it's a clear win.

I'll play around with this in a follow up PR.